src/wovnio/wovnphp/Store.php

```diff
@@ -125,7 +126,13 @@ public function updateSettings($vals)
 
         // getting the url pattern
         if (isset($vals['url_pattern_name']) && $vals['url_pattern_name'] === 'query') {
-            $vals['url_pattern_reg'] = '((\?.*&)|\?)wovn=(?P<lang>[^&]+)(&|$)';
+            if (isset($vals['lang_param_name'])) {
+                $lang_param_name = $vals['lang_param_name'];
+            } else {
+                $defaultSettings = $this->defaultSettings();
```

Comment:
`$this->defaultSettings()` are already loaded down in `updateSettings` so you should instanciate `$defaultSettings` at top scope for single call.

Reply:
done

I was hesitant to initialize `$defaultSettings` at the top of the function because accessing it feels like a hack with the way settings are handled in Store.php at the moment. But you're right, it's just more dumb to initialize it twice. =/

Reply:
It does feel like a hack, yes.